### ISSUE TITLE
Option to block all requests to specific endpoints

### ISSUE THREAD

**Issue by psiinon** [zaproxy/zaproxy#8686]
### Is your feature request related to a problem? Please describe.

We can exclude URL regex patterns in various ways.
We can also block proxied requests on any basis via scripts e.g. [Drop requests by response code.js](https://github.com/zaproxy/community-scripts/blob/main/proxy/Drop%20requests%20by%20response%20code.js)
However we cannot currently easily block requests from ZAP tools (like the spiders or active scan) in a flexible way.
E.g. if OpenAPI definition includes a GET and DELETE method on the same URL then its hard to just block the DELETE and not the GET.

### Describe the solution you'd like:

An easier way to stop the ZAP tools from accessing specific endpoints.
This could be by adding more options to the `excludeFrom` options, or by making it possible to block requests in httpsender scripts.
Other suggestions appreciated 😁 

### Describe alternatives you've considered:

The only current option is to create an httpsender script and to change the URL to one that is known to fail.
Not very pretty, but that should work.

### Would you like to help fix this issue?

- [ ] Yes

**Comment by psiinon** [comment]
Related issue: https://github.com/zaproxy/zaproxy/issues/3121

**Comment by kingthorin** [comment]
I'm going to self assign this, but likely just to write a community script to provide a temporary workaround/option.

**Comment by thc202** [comment]
I don't think we should be adding that to the community scripts, especially since this is not asked that often.

If this is implemented for the HTTP Sender listeners we should be careful to keep the same expectations to avoid breaking existing code/logic.
